weblate/api/views.py

```diff
@@ -253,6 +256,115 @@ def repository(self, request, **kwargs):
         return Response(data)
 
 
+class UserViewSet(viewsets.ModelViewSet):
+    """Users API."""
+
+    queryset = User.objects.none()
+    serializer_class = UserSerializer
+    lookup_field = "username"
+
+    def get_queryset(self):
+        if self.request.user.has_perm("user.edit"):
+            return User.objects.order_by("id").all()
+        return User.objects.filter(pk=self.request.user.pk).order_by("id").all()
+
+    def perm_check(self, request):
+        if not request.user.has_perm("user.edit"):
+            self.permission_denied(request, message="Can not manage Users")
+
+    def update(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().update(request, *args, **kwargs)
+
+    def create(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().create(request, *args, **kwargs)
+
+    def destroy(self, request, *args, **kwargs):
+        self.perm_check(request)
+        instance = self.get_object()
+        remove_user(instance, request)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def partial_update(self, request, *args, **kwargs):
```

Comment:
Is this method needed? It should be already available in ModelViewSet. It calls update under the hood, so the permission check is covered as well.